crates/trie/sparse/src/trie.rs

```diff
@@ -3616,6 +3641,41 @@ mod tests {
         assert_eq!(sparse.root(), EMPTY_ROOT_HASH);
     }
 
+    #[test]
+    fn sparse_trie_clear() {
+        // tests that if we fill a sparse trie with some nodes and then clear it, it has the same
+        // contents as an empty sparse trie
+        let mut sparse = RevealedSparseTrie::default();
+        let value = alloy_rlp::encode_fixed_size(&U256::ZERO).to_vec();
+        sparse
+            .update_leaf(Nibbles::from_nibbles([0x5, 0x0, 0x2, 0x3, 0x1]), value.clone())
+            .unwrap();
+        sparse
+            .update_leaf(Nibbles::from_nibbles([0x5, 0x0, 0x2, 0x3, 0x3]), value.clone())
+            .unwrap();
+        sparse
+            .update_leaf(Nibbles::from_nibbles([0x5, 0x2, 0x0, 0x1, 0x3]), value.clone())
+            .unwrap();
+        sparse.update_leaf(Nibbles::from_nibbles([0x5, 0x3, 0x1, 0x0, 0x2]), value).unwrap();
+
+        sparse.clear();
+
+        // we have to update the root hash to be an empty one, because the `Default` impl of
+        // `RevealedSparseTrie` sets the root hash to `EMPTY_ROOT_HASH` in the constructor.
+        //
+        // The default impl is only used in tests.
+        sparse.nodes.insert(Nibbles::default(), SparseNode::Empty);
+
+        let empty_trie = RevealedSparseTrie::default();
+        assert_eq!(sparse.nodes, empty_trie.nodes);
+        assert_eq!(sparse.values, empty_trie.values);
+        assert_eq!(sparse.branch_node_hash_masks, empty_trie.branch_node_hash_masks,);
+        assert_eq!(sparse.branch_node_tree_masks, empty_trie.branch_node_tree_masks,);
+        assert_eq!(sparse.prefix_set, empty_trie.prefix_set);
+        assert_eq!(sparse.updates, empty_trie.updates);
+        assert_eq!(sparse.rlp_buf, empty_trie.rlp_buf);
```

Comment:
can we do a comparison of the whole object? otherwise if we add another field, we will as well forget to fix the test 😄

Reply:
oh, true lol yeah